codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -161,29 +161,17 @@ public B gracefulShutdownTimeoutMillis(long gracefulShutdownTimeoutMillis) {
         }
 
         /**
-         * Determine if the encoder should queue frames to honor the value set by
-         * {@link #encoderMaxConcurrentStreams(int)}.
-         */
-        public B encoderEnforceMaxConcurrentStreams(boolean encoderEnforceMaxConcurrentStreams) {
-            encoderMaxConcurrentStreams = -1;
-            return thisB();
-        }
-
-        private boolean encoderEnforceMaxConcurrentStreams() {
-            return encoderMaxConcurrentStreams >= 0;
-        }
-
-        /**
          * How many initial streams are allowed to exists concurrently. Frames will be queued if they would result in
          * creating a stream which would cause the number of existing streams to exceed this number.
-         * @see #encoderEnforceMaxConcurrentStreams(boolean)
+         *
+         * Use {@code -1} if you not want to enforce max concurrent streams.
          */
         public B encoderMaxConcurrentStreams(int encoderMaxConcurrentStreams) {
             // This bounds are enforced here because the builder makes assumptions about its valid range to determine
             // if it should be used.
-            if (encoderMaxConcurrentStreams < 0) {
+            if (encoderMaxConcurrentStreams < -1) {
```

Comment:
@Scottmitch We can't use the initialSettings because that specifies the max concurrent streams allowed to be created by the remote endpoint (we need the opposite). 

I wonder if we need the `encoderBuffer` property at all ... instead just always use the 1-arg constructor for `StreamBufferingEncoder`, which assumes 100 streams.  The HTTP/2 spec suggests that all endpoints support 100 streams as a lower bound, so that's the default behavior and is probably reasonable for just about any implementation. If someone asks for this to be configured we can come back to this later.

WDYT?